vnu-java-resolver.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const { spawnSync, execSync } = require('child_process');
+const { mkdirSync, existsSync, createWriteStream } = require('fs');
+const { join, dirname, parse } = require('path');
+const { extract } = require('tar');
+const { Readable } = require('stream');
+
+function findNearestNodeModules(startDir) {
+    let dir = startDir;
+    const { root } = parse(startDir);
+    while (true) {
+        const candidate = join(dir, 'node_modules');
+        if (existsSync(candidate)) {
+            return candidate;
+        }
+        if (dir === root) {
+            throw new Error('Could not find a node_modules directory.');
+        }
+        dir = dirname(dir);
+    }
+}
+
+const NODE_MODULES_DIR = findNearestNodeModules(__dirname);
+const CACHE_DIR = join( NODE_MODULES_DIR, '.cache', 'vnu-jar', 'java');
+const TEMURIN_BASE_URL = 'https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.17%2B10/';
+
+function findSystemJava() {
+    const whichCmd = process.platform === 'win32' ? 'where' : 'which';
+    const result = spawnSync(whichCmd, ['java'], { encoding: 'utf8' });
+    if (result.status === 0 && result.stdout) {
+        return result.stdout.split(/\r?\n/)[0].trim();
+    }
+    return null;
+}
+
+function getPlatformArchiveName() {
+    const arch = process.arch;
+    const plat = process.platform === 'win32' ? 'windows' : process.platform === 'darwin' ? 'mac' : 'linux';
+    const ext = process.platform === 'win32' ? 'zip' : 'tar.gz';
+    return `OpenJDK17U-jre_${arch}_${plat}_hotspot_17.0.17_10.${ext}`;
+}
+
+async function downloadJava() {
+    mkdirSync(CACHE_DIR, { recursive: true });
+    const archiveName = getPlatformArchiveName();
+    const url = TEMURIN_BASE_URL + archiveName;
+    const archivePath = join(CACHE_DIR, archiveName);
+    console.log(`Downloading Java 17 runtime from ${url}...`);
+    const res = await fetch(url);
+    if (!res.ok) {
+        throw new Error(`Failed to download Java: ${res.status} ${res.statusText}`);
+    }
```

Comment:
On my Macbook M1 Air, I got this error:
```
Downloading Java 17 runtime from https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.17%2B10/OpenJDK17U-jre_arm64_mac_hotspot_17.0.17_10.tar.gz...
Error: Failed to download Java: 404 Not Found
```

archiveName: `OpenJDK17U-jre_arm64_mac_hotspot_17.0.17_10.tar.gz`